source/index.ts

```diff
@@ -19,7 +22,7 @@ const validFilename = require('valid-filename')
  * @returns {void}
  * @private
  */
-function validator(fullPath, relativePath) {
+function validator(this: any, fullPath: string, relativePath: string) {
```

Comment:
no any cheating, what does this `any` represent? it is the same type being passed around in several other instances, using a type alias will help